### PR TITLE
Alias Expiration

### PR DESCRIPTION
**Pull request by frankTheTank72**

### CODE REVIEW

File: SIP/sip-48.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+The following changes should be done with an upcoming hard-fork to introduce the renewal process for aliases.
+
+### Database changes
+On the database level, the alias table will get a new field `renewal_time` which is an INT field and contains the timestamp until the alias is reserved for the current owner. With introduction of the needed hard-fork all aliases will get a reserved datetime of 3 months after the planned hard fork. In this case all active users have enough time to check their current alias ownerships and can create a renewal subscription if desired.
+
+### Node changes
+**Transaction Set Alias**
+
+If a user sends a setAlias transaction the following datetime handling for the alias will be executed:
+ - If the creation is before the block of the hard fork no datetime is set on `renewal_time`.
```
Comment (jjos2372) [inline SIP/sip-48.md:27]: These new transactions cannot be sent before the fork, I believe all this logic for before the fork block should be removed. These transactions simply are invalid before that.

File: SIP/sip-48.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ - If the creation is after the hard fork and no datetime is set for the alias or a datetime which is in the past or shorter than 24 hours in the future, a new datetime will be set with datetime plus 24 hours on `renewal_time`.
+ - If the creation is after the hard fork but  the datetime is longer than 24 hours in the future no datetime adjustment will happen on `renewal_time`
+
+In addition, if the setAlias transaction is executed for an existing alias and the creator of the setAlias transaction is not the current owner, the transaction - which leads to an automated owner transfer -  will only be executed if the renewal timestamp of the alias has already expired.
+
+**Auto renewal for aliases**
```
Comment (jjos2372) [inline SIP/sip-48.md:33]: I'm not in favor of this proposal, it mixes contract logic with native services. Plus, can be source of criticism on how these funds are distributed. I would remove this and people can simply run a script if they want to renew.

File: SIP/sip-48.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- **receiver** is equal the staking contract
+- **frequency** is equal 1 year in seconds
+- **amount** is equal 70 Signa
+- **alias** should be set with the aliasId to renewal - with this attribute the SIP-44 message will be created
+
+The transaction created with this call is still the same as the standard subscription creation.
+
+
+###  Renewal costs
+The renewal costs are set with 70 Signa. The reason for this amount is given by the alternative that a user needs to execute setAlias every day to update the `reneval_time` by 24 hours. This would generate costs of 365 x 0.2 Signa = 73 Signa.
```
Comment (jjos2372) [inline SIP/sip-48.md:59]: How this compares to getting a new alias?

File: SIP/sip-48.md (hunk 2 above)
Comment (ohager) [reply SIP/sip-48.md:34]: Same here. The separation between protocol layer and logic layer has to be strict!

**Review by ohager — CHANGES_REQUESTED** [review]
I'm not in favor of this proposal. Yes, Alias expiration is a good thing, and for businesses it is essential to have a robust/reliable renewal option. This proposal mixes up protocol and logic layers and even wants to tie up a token into this, which is problematic because of:

- planned multichain
- favoritism issues 

My suggestion is to introduce either a new Subscription Subtype `Renewal Subscription` or allow `Subscription` to have an additional alias Id parameter, thus linking the subscription to an alias renewal. It could be even interesting to allow flexible renewal periods, e.g. 1 - 24 months, and a renewal fee according to this. 

In my opinion, this solution is a pure protocol based approach, which is even easy to handle by the users.

File: SIP/sip-48.md (hunk 2 above)
Comment (frankTheTank72) [reply SIP/sip-48.md:34]: As i said - we set a receiver on the subscription to be valid for an renewal of an alias. We could also set this receiver to Zero to be valid aka burning - but i think we should re-distributed this payments to the community and noz burn them

**Comment by frankTheTank72** [comment]
> I'm not in favor of this proposal. Yes, Alias expiration is a good thing, and for businesses it is essential to have a robust/reliable renewal option. This proposal mixes up protocol and logic layers and even wants to tie up a token into this, which is problematic because of:
> 
> 
> 
> - planned multichain
> 
> - favoritism issues 
> 
> 
> 
> My suggestion is to introduce either a new Subscription Subtype `Renewal Subscription` or allow `Subscription` to have an additional alias Id parameter, thus linking the subscription to an alias renewal. It could be even interesting to allow flexible renewal periods, e.g. 1 - 24 months, and a renewal fee according to this. 
> 
> 
> 
> In my opinion, this solution is a pure protocol based approach, which is even easy to handle by the users.


I belief that a combination of existing protocol features can just be combined to handle the renewal - the only new thing needed, is to handle the renewal_time on the alias. 

I also belief that a simple 1 year  subscription is enough to handle the whole workflow - no need to have customized periods - as this is also not typical for domain registrations. In addition, we provide a simple APi call with fixed values so a subscription creation is seamless for the user.

File: SIP/sip-48.md (hunk 2 above)
Comment (jjos2372) [reply SIP/sip-48.md:34]: My comment was regarding mixing the protocol with a specific contract. I have no problem in extending the protocol to support somehow automatic alias renewal.

**Comment by ohager** [comment]
> > I'm not in favor of this proposal. Yes, Alias expiration is a good thing, and for businesses it is essential to have a robust/reliable renewal option. This proposal mixes up protocol and logic layers and even wants to tie up a token into this, which is problematic because of:
> > 
> > * planned multichain
> > * favoritism issues
> > 
> > My suggestion is to introduce either a new Subscription Subtype `Renewal Subscription` or allow `Subscription` to have an additional alias Id parameter, thus linking the subscription to an alias renewal. It could be even interesting to allow flexible renewal periods, e.g. 1 - 24 months, and a renewal fee according to this.
> > In my opinion, this solution is a pure protocol based approach, which is even easy to handle by the users.
> 
> I belief that a combination of existing protocol features can just be combined to handle the renewal - the only new thing needed, is to handle the renewal_time on the alias.
> 
> I also belief that a simple 1 year subscription is enough to handle the whole workflow - no need to have customized periods - as this is also not typical for domain registrations. In addition, we provide a simple APi call with fixed values so a subscription creation is seamless for the user.

The customizable prolongation period was just a suggestion. I have no problem to fix it to a single option only. This would make it even easier to implement.

File: SIP/sip-48.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+status: Review
+type: standard
+category: Core
+created: 2022-12-10
+---
+## Abstract
+This SIP will introduce a temporary ownership of aliases which can be prolonged on a yearly basis.
+
+## Motivation
+Signum offers a simple setup to create aliases for an account. Once set/created those aliases are owned by the creator account forever. With the introducing of [SIP-44](sip-44.md)  the alias will be more powerful to use for like domain, account  or application pointer which makes a reserved alias name more valuable. In addition with [SIP-47](sip-44.md)  a new service SNS (Signum Naming Service) will be introduced which is a distributed, open, and extensible naming system based on the Signum blockchain. Given these cases, those aliases should be actively maintained by the creator/owner to flag those aliases are in use by them, otherwise another user should be able to claim the alias.
```
Comment (ohager) [inline SIP/sip-48.md:15]: `...with the introduction` 
several doubled spaces

File: SIP/sip-48.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ - If the creation is after the hard fork but  the datetime is longer than 24 hours in the future no datetime adjustment will happen on `renewal_time`
+ - If the setAlias leads to a new alias the `renewal_time` is set with a value of 24 hours in the future.
+
+In addition, if the setAlias transaction is executed for an existing alias and the creator of the setAlias transaction is not the current owner, the transaction - which leads to an automated owner transfer -  will only be executed if the renewal timestamp of the alias has already expired.
+
+**Auto renewal for aliases**
+
+Signum has already all the basic functionality to empower the user for an auto-renewal via the existing subscription transaction. A subscription is valid for the auto-renewal when the following parameters are given:
+
+- The subscription is created with an attachment. A new field `alias` need to be set to bind the subscription to a given alias.
```
Comment (ohager) [inline SIP/sip-48.md:38]: :+1:

File: SIP/sip-48.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+- The subscription amount is paid to a defined receiver account.
+- The subscription should have minimum amount of 70 SIGNA as payment.
+- The subscription should be set to a 1 year interval (in seconds).
+
+**New node logic by execution of a subscription:**
+
+By creation or an interval payment of the subscription the node will check the above parameters. If the creator of the subscription is still the owner of the alias and the balance of the account sufficient to execute the subscription, the alias will get an update on the  `renwal_time` table field by one year and 24 hours (based on the execution time of the subscription). If the creator is no longer the owner of the alias the subscription will be cancelled before any further execution.
```
Comment (ohager) [inline SIP/sip-48.md:45]: doubled spaces

File: SIP/sip-48.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+- **amount** is equal 70 Signa
+- **alias** should be set with the aliasId to renewal - with this attribute the attachment will be created.
+
+The transaction created with this call is still the same as the standard subscription creation.
+
+
+###  Renewal costs
+The renewal costs are set with 70 Signa. The reason for this amount is given by the alternative that a user needs to execute setAlias every day to update the `reneval_time` by 24 hours. This would generate costs of 365 x 0.2 Signa = 73 Signa. 
+
+### Mulitverse support
```
Comment (ohager) [inline SIP/sip-48.md:62]:
```suggestion
### Multiverse support
```

If the receiver is defined to be burn address, this section could be omitted

**Review by ohager — APPROVED** [review]

**Pull request merged**